VRDR.HTTP/Program.cs

```diff
@@ -47,57 +44,80 @@ public static string SendResponse(HttpListenerRequest request)
-                case string xmlType when new Regex(@"xml").IsMatch(xmlType): // application/fhir+xml
-                default:
-                    deathRecord = new DeathRecord(requestBody);
-                    break;
+                throw new Exception(e.Message);
             }
 
             // Look at URL extension to determine output format; be permissive in what we accept as format specification
             string result = "";
-            switch (request.RawUrl)
+            try
             {
-                case string url when new Regex(@"(ije|mor)$").IsMatch(url): // .mor or .ije
-                    IJEMortality ije = new IJEMortality(deathRecord);
-                    result = ije.ToString();
-                    break;
-                case string url when new Regex(@"json$").IsMatch(url): // .json
-                    result = deathRecord.ToJSON();
-                    break;
-                case string url when new Regex(@"xml$").IsMatch(url): // .xml
-                    result = deathRecord.ToXML();
-                    break;
-                case string url when new Regex(@"nightingale$").IsMatch(url): // .nightingale
-                    result = Nightingale.ToNightingale(deathRecord);
-                    break;
+                switch (request.RawUrl)
+                {
+                    case string url when new Regex(@"(ije|mor)$").IsMatch(url): // .mor or .ije
+                        IJEMortality ije = new IJEMortality(deathRecord);
+                        result = ije.ToString();
+                        break;
+                    case string url when new Regex(@"json$").IsMatch(url): // .json
+                        result = deathRecord.ToJSON();
+                        break;
+                    case string url when new Regex(@"xml$").IsMatch(url): // .xml
+                        result = deathRecord.ToXML();
+                        break;
+                    case string url when new Regex(@"nightingale$").IsMatch(url): // .nightingale
+                        result = Nightingale.ToNightingale(deathRecord);
+                        break;
+                }
+            }
+            catch (Exception e)
+            {
+                throw new Exception(e.Message);
             }
 
             return result;
         }
 
         public static string GetBodyContent(HttpListenerRequest request)
         {
-            using (System.IO.Stream body = request.InputStream)
+            using (Stream body = request.InputStream)
             {
-                using (System.IO.StreamReader reader = new System.IO.StreamReader(body, request.ContentEncoding))
+                using (StreamReader reader = new StreamReader(body, request.ContentEncoding))
                 {
                     return reader.ReadToEnd();
                 }
             }
         }
 
+        public static String GenerateJsonResponse(string type, string data)
+        {
+            var response = "{" +

```

Comment:
Manually building the JSON like this could produce invalid JSON if, e.g., the error in the exception happened to contain quotes. I'll suggest using something like

```
JsonConvert.SerializeObject(new { "type" = type, "detail", data }) // I didn't test this code
```

instead. Given that it's a single line of code, there's probably also no need to wrap it in a GenerateJsonResponse method.